src/compiler/checker.ts

```diff
@@ -39802,8 +39806,35 @@ namespace ts {
                     }
                 }
 
-                if (isImportSpecifier(node) && target.declarations?.every(d => !!(getCombinedNodeFlags(d) & NodeFlags.Deprecated))) {

```

Comment:
this doesn't call isUncalledFunctionReference before, but now checkDeprecatedAliasedSymbol will. Is that OK? Does isUncalledFunctionReference have a quick exit for non-functions?

Reply:
@sandersn Thanks for the review. Yep, `isUncalledFunctionReference` is required for use in `checkIdentifier`. This does not affect the `ImportSpecifier` check, but in this case, it is a useless condition. I've changed `checkDeprecatedAliasedSymbol` to return the target of the alias. Does that make sense?